File: backend/src/main/java/edu/cit/amihan/medibook/appointment/dto/AppointmentResponse.java
package edu.cit.amihan.medibook.appointment.dto;

import edu.cit.amihan.medibook.appointment.entity.Appointment;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AppointmentResponse {

    private Long appointmentId;
    private Long patientId;
    private String patientName;
    private Long doctorId;
    private String doctorName;
    private LocalDateTime startTime;
    private LocalDateTime endTime;
    private String status;
    private LocalDateTime createdAt;

    public static AppointmentResponse fromEntity(Appointment appointment) {
        return AppointmentResponse.builder()
                .appointmentId(appointment.getAppointmentId())
                .patientId(appointment.getPatient().getPatientId())
                .patientName(appointment.getPatient().getFullName())
                .doctorId(appointment.getSchedule().getDoctor().getDoctorId())
                .doctorName(appointment.getSchedule().getDoctor().getFullName())
                .startTime(appointment.getSchedule().getStartTime())
                .endTime(appointment.getSchedule().getEndTime())
                .status(appointment.getStatus().name())
                .createdAt(appointment.getCreatedAt())
                .build();
    }
}

File: mobile/app/src/main/java/edu/cit/amihan/medibook/feature/appointment/model/AppointmentResponse.kt
package edu.cit.amihan.medibook.feature.appointment.model

import com.google.gson.annotations.SerializedName

data class AppointmentResponse(
    @SerializedName("appointmentId") val appointmentId: Long = 0L,
    @SerializedName("patientId") val patientId: Long = 0L,
    @SerializedName("patientName") val patientName: String? = null,
    @SerializedName("doctorId") val doctorId: Long = 0L,
    @SerializedName("doctorName") val doctorName: String? = null,
    @SerializedName("startTime") val startTime: String? = null,
    @SerializedName("endTime") val endTime: String? = null,
    @SerializedName("status") val status: String? = null,
    @SerializedName("createdAt") val createdAt: String? = null
)